marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/flow/jobQueryLib.sjs

```diff
@@ -18,36 +18,46 @@
 const op = require('/MarkLogic/optic');
 const config = require("/com.marklogic.hub/config.sjs");
 
-function findStepResponses(start, pageLength, sortColumn, sortDirection) {
+function findStepResponses(query) {
+  const start = query.start;
+  const pageLength = query.pageLength;
+  const sortColumn = query.sortColumn;
+  const sortDirection = query.sortDirection;
+
   let orderByConstraint = op.desc('startTime');
-  if(sortColumn) {
+  if(query.sortColumn) {
     orderByConstraint = sortDirection === 'ascending' ? op.asc(sortColumn) : op.desc(sortColumn);
   }
 
-  const response = {};
-  const totalCountQuery = 'select count(*) as total from Job.StepResponse';
-  const jobsDataQuery = 'select Job.StepResponse.stepName as stepName,' +
-      'Job.StepResponse.stepDefinitionType as stepDefinitionType,' +
-      'Job.StepResponse.stepStatus as status,' +
-      'Job.StepResponse.targetEntityType as entityName,' +
-      'Job.StepResponse.stepStartTime as startTime,' +
-      'Job.StepResponse.stepEndTime - Job.StepResponse.stepStartTime as duration,' +
-      'Job.StepResponse.successfulItemCount as successfulItemCount,' +
-      'Job.StepResponse.failedItemCount as failedItemCount,' +
-      'Job.StepResponse.user as user,' +
-      'Job.StepResponse.jobId as jobId,' +
-      'Job.StepResponse.flowName as flowName' +
-      ' from Job.StepResponse';
-
-  const totalCount = op.fromSQL(totalCountQuery).result().toObject();
-  const jobsDataResults = op.fromSQL(jobsDataQuery).orderBy(orderByConstraint).offset(pageLength * (start-1)).limit(pageLength).result();
-
-  response["total"] = totalCount[0]["total"];
-  response["start"] = start;
-  response["pageLength"] = pageLength;
-  response["results"] = jobsDataResults.toObject();
-
-  return response;
+  return fn.head(xdmp.invokeFunction(
```

Comment:
I don't think there's a use case for invokeFunction here. This is an internal API, and we know we want to use the DatabaseClient for the jobs database when we do this. So I think the way you had it before was better.